components/recommendation/RecommendationArticle.vue

```diff
@@ -0,0 +1,63 @@
+<script setup lang="ts">
+import type { Recommendation } from '../composables/recommendations'
+
+const {
+  item,
+} = defineProps<{
+  item: Recommendation
+}>()
+
+const target = ref<HTMLDivElement>()
+const { isActive } = useIntersectionObserver(target, checkIntersection)
+
+function checkIntersection([{ isIntersecting }]: [{ isIntersecting: boolean }]): void {
```

Comment:
this line hurts to dissect at first but got better afterwards 😃

Reply:
I don't like it, too. I could probably add some line breaks to make it a little easier to grok.